src/sources/kubernetes_logs/reconciler.rs

```diff
@@ -0,0 +1,302 @@
+        }
+    }
+}
+
+#[derive(Clone)]
+struct EventStreamBuilder {
+    client: Client,
+    line_sender: mpsc::Sender<Vec<Line>>,
+}
+
+#[derive(Clone)]
+enum TailerState {
+    Running,
+}
+
+impl EventStreamBuilder {
+    pub fn start(&self, container_info: ContainerInfo) -> TailerState {
+        let this = self.clone();
+        tokio::spawn(async move {
+            let log_info = ContainerLogInfo::new(&container_info, Utc::now());
+            this.run_event_stream(log_info).await;
+        });
+        TailerState::Running
+    }
+
+    pub async fn run_event_stream(mut self, log_info: ContainerLogInfo<'_>) {
+        let pods: Api<Pod> =
+            Api::namespaced(self.client.clone(), &log_info.container_info.namespace);
+
+        info!(
+            "Starting streaming log tail for container '{}' in pod '{}' (namespace '{}') from timestamp {}",
+            log_info.container_info.container_name,
+            log_info.container_info.pod_name,
+            log_info.container_info.namespace,
+            log_info.log_since()
+        );
+
+        let log_params = LogParams {
+            container: Some(log_info.container_info.container_name.clone()),
+            follow: true,
+            since_time: Some(log_info.log_since()),
+            timestamps: true,
+            ..Default::default()
+        };
+
+        match pods
+            .log_stream(&log_info.container_info.pod_name, &log_params)
+            .await
+        {
+            Ok(log_stream) => {
+                info!(
+                    "Started streaming logs from container '{}' in pod '{}'",
+                    log_info.container_info.container_name, log_info.container_info.pod_name
+                );
+
+                let mut buffer = Vec::new();
+                let mut log_stream = log_stream;
+
+                // Process the stream by reading line by line
+                loop {
+                    match log_stream.read_until(b'\n', &mut buffer).await {
+                        Ok(0) => break, // EOF
+                        Ok(_) => {
+                            // Remove trailing newline if present
+                            if buffer.ends_with(b"\n") {
+                                buffer.pop();
+                            }
+                            // Remove trailing carriage return if present (for CRLF)
+                            if buffer.ends_with(b"\r") {
+                                buffer.pop();
+                            }
+
+                            let line_bytes = Bytes::from(std::mem::take(&mut buffer));
+
+                            // TODO: track last log timestamp
+
+                            let text_len = line_bytes.len() as u64;
+                            let line = Line {
+                                text: line_bytes,
+                                filename: String::new(), // Filename is not applicable for k8s logs
+                                file_id: FileFingerprint::FirstLinesChecksum(0),
```

Comment:
This is not correct and will likely cause issues with the file source implementation